lib/shoulda/matchers/action_controller/flash_store.rb

```diff
@@ -36,7 +36,7 @@ def has_value?(expected_value)
         end
 
         def empty?
-          flash.empty?
+          flash&.empty?
```

Comment:
I'm curious why this is necessary. Are there issues you noticed when using the gem?

Reply:
This is a case of Rails delegation. The `empty?` method is being delegated to the `flush` object. We can replace the `empty?` method by either `delegate :empty?, to: :flush` or we can add the safe navigation operator as the Rails' `allow_nil` option checks not just for nil but also delegates if nil responds to the delegated method.

Reply:
I see. So, `&.` is a handy operator, but you have to keep in mind that it's really a form of defensive coding. What you're saying when you're using it is, "I expect at some point for this value to be nil, so if that happens, do this." Sometimes defensive coding is perfectly fine because you might have a method you wrote and you know that there is a chance it could return nil (because you wrote it that way). In this case, and in the other files you've changed here, I don't think it makes sense to use defensive coding. We don't want `flash`, `attribute_setter`, etc. to be nil, and there's no reason we ought to expect it to be nil. If nil pops up, that's an indication of a larger problem that should be addressed at a higher level. Adding `&.`, in my opinion, sweeps that problem under the rug, when it should be blaring. Therefore, I'm not sure we need these `&.` changes.

Reply:
Makes sense. Since we do not expect the `flash` object to be nil ever, and using the `&.` safe navigation operator will hide any errors, we can instead use the other method suggested above to perform Rails delegations, as so: `delegate :empty?, to: :flash` instead.

Reply:
Yup, that's a good point.